### PR TITLE
Auto apply constants for oneOf selection

### PR DESCRIPTION
**Pull request by Logende**

### CODE REVIEW

File: frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue [hunk 1]
```diff
@@ -31,14 +35,60 @@ const valueProperty = computed({
...
+  // of the property with the selected oneOf.
+  // If the merged schema contains constants, those are applied to the data.
+  // Note that it might seem logical to compute this merged schema only once
+  // (e.g. when oneOf is selected), store it and re-use it later.
+  // This, however, is not done because the effective merged schema could change
+  // even without a new oneOf selection by the user, due to JSON schema features
+  // such as if and else.
+
+  // TODO: maybe this function should instead of normal schema make use of the effective schema
+  const schemaAtPath = useSessionStore().schemaAtPath(path);
```
Comment (PaulBredl) [inline frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue:65]: yes it should use the effective schema

File: frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue [hunk 2]
```diff
@@ -31,14 +35,60 @@ const valueProperty = computed({
...
+    baseSchema,
+    schemaAtPath.oneOf[selectedOneOfOption.index].jsonSchema
+  );
+  const resultData = applySchemaConstantsOnData(mergedSchema, props.propertyData);
+  if (!_.isEqual(resultData, props.propertyData)) {
+    emit('update_property_value', resultData);
+  }
+}
+function applySchemaConstantsOnData(schema: JsonSchemaObjectType, data: any): any {
+  // note that in pre-processing all const is converted to an enum with just one entry
```
Comment (Logende) [inline frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue:78]:
```suggestion
  // note that in pre-processing all const is converted to an enum with just one entry
  // hence, for us constants are equal to an enum of length 1.
```

**Comment by Logende** [comment]
> I see that feature a bit critical in general, but we can try it out and see how it feels over time

I'm also thinking about using it only for oneOf and not anyOf and especially with the use-case of creating new properties in the schema editor in mind. If the even more simplified meta schema chooses a different solution (e.g. oneOf of simpleProperty, object, reference and Blabla and then within simpleProperty the user chooses the type and then if-else make sure the correct properties are shown for the chosen type) then maybe we can drop this. But maybe it still makes life of the user easier, I'd just try it out.

File: frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue (hunk 1 above)
Comment (PaulBredl) [reply frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue:65]: In fact, we should probably make a warning to `schemaAtPath` that `effectiveSchemaAtPath` should be used in most cases

Comment (Logende) [reply frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue]: fixed

**Comment by Logende** [comment]
#308

**Review by PaulBredl — APPROVED** [review]

**Pull request merged**